src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -427,11 +427,11 @@ abstract class RefChecks extends Transform {
                                           "(this rule is designed to prevent accidental overrides)")
             } else if (other.isStable && !member.isStable) { // (1.4)
               overrideErrorWithMemberInfo("stable, immutable value required to override:")
-            } else if (member.isValue && member.isLazy &&
+            } else if (member.isValue && (member.isLazy || member.isModule) &&
```

Comment:
Could we make this a deprecation warning for now, and error under `-Xsource:3`?

Reply:
I was scared of breaking the matrix of exclusive `else ifs`.